internal/p2p/peermanager.go

```diff
@@ -1273,11 +1279,18 @@ func (s *peerStore) loadPeers() error {
 			return fmt.Errorf("invalid peer data: %w", err)
 		}
 		peers[peer.ID] = peer
+		for addr := range peer.AddressInfo {
+			// TODO maybe check to see if we've seen this
+			// addr before for a different peer, there
+			// could be duplicates.
+			addrs[addr] = peer.ID
```

Comment:
Do we necessarily want to replace an existing peer ID for this address, though?

Reply:
I think this shouldn't ever happen, *but* it's basically impossible to know what the right answer is here *and* I think people might have junk addresses in the peer database. 

I think the way we use this index, though, is safe because we don't want to dial an address t hat we know we're connected to already or trying to dial: we might miss dialing good addresses if we're hanging on dialing a bad address, but we never want to try both at once.